modules/3d/samples/odometry_evaluation.cpp

```diff
@@ -0,0 +1,245 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+#include <opencv2/3d.hpp>
+
+#include <opencv2/highgui.hpp>
+#include <opencv2/3d.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/core/utility.hpp>
+
+#include <iostream>
+#include <fstream>
+
+using namespace std;
+using namespace cv;
+
+#define BILATERAL_FILTER 0// if 1 then bilateral filter will be used for the depth
+
+class MyTickMeter
+{
+public:
+    MyTickMeter() { reset(); }
+    void start() { startTime = getTickCount(); }
+    void stop()
+    {
+        int64 time = getTickCount();
+        if ( startTime == 0 )
+            return;
+        ++counter;
+        sumTime += ( time - startTime );
+        startTime = 0;
+    }
+
+    int64 getTimeTicks() const { return sumTime; }
+    double getTimeSec()   const { return (double)getTimeTicks()/getTickFrequency(); }
+    int64 getCounter() const { return counter; }
+
+    void reset() { startTime = sumTime = 0; counter = 0; }
+private:
+    int64 counter;
+    int64 sumTime;
+    int64 startTime;
+};
+
+static
+void writeResults( const string& filename, const vector<string>& timestamps, const vector<Mat>& Rt )
```

Comment:
What format is used?

Why is it not a `FileStorage`?

Reply:
The format used here is for [RGBD SLAM benchmarking](https://vision.in.tum.de/data/datasets/rgbd-dataset/online_evaluation), it's described [here](https://vision.in.tum.de/data/datasets/rgbd-dataset/file_formats).